src/java.base/share/classes/jdk/internal/classfile/components/snippet-files/PackageSnippets.java

```diff
@@ -0,0 +1,212 @@
+                else
+                    classBuilder.accept(classElement);
+            });
+        // @end
+    }
+
+    void codeRelabeling(ClassModel classModel) {
+        // @start region="codeRelabeling"
+        byte[] newBytes = classModel.transform(
+                ClassTransform.transformingMethodBodies(
+                        CodeTransform.ofStateful(CodeRelabeler::of)));
+        // @end
+    }
+
+    // @start region="classInstrumentation"
+    byte[] classInstrumentation(ClassModel target, ClassModel instrumentor, Predicate<MethodModel> instrumentedMethodsFilter) {
+        var instrumentorCodeMap = instrumentor.methods().stream()
+                                              .filter(instrumentedMethodsFilter)
+                                              .collect(Collectors.toMap(mm -> mm.methodName().stringValue() + mm.methodType().stringValue(), mm -> mm.code().orElse(null)));
+        var targetFieldNames = target.fields().stream().map(f -> f.fieldName().stringValue()).collect(Collectors.toSet());
+        var targetMethods = target.methods().stream().map(m -> m.methodName().stringValue() + m.methodType().stringValue()).collect(Collectors.toSet());
+        var instrumentorClassRemapper = ClassRemapper.of(Map.of(instrumentor.thisClass().asSymbol(), target.thisClass().asSymbol()));
+        return target.transform(
+                ClassTransform.transformingMethods(
+                        instrumentedMethodsFilter,
+                        (mb, me) -> {
+                            if (me instanceof CodeModel targetCodeModel) {
+                                var mm = targetCodeModel.parent().get();
+                                //instrumented methods code is taken from instrumentor
+                                mb.transformCode(instrumentorCodeMap.get(mm.methodName().stringValue() + mm.methodType().stringValue()),
+                                        //all references to the instrumentor class are remapped to target class
+                                        instrumentorClassRemapper.asCodeTransform()
+                                        .andThen((codeBuilder, instrumentorCodeElement) -> {
+                                            //all invocations of target methods from instrumentor are inlined
+                                            if (instrumentorCodeElement instanceof InvokeInstruction inv
+                                                && target.thisClass().asInternalName().equals(inv.owner().asInternalName())
+                                                && mm.methodName().stringValue().equals(inv.name().stringValue())
+                                                && mm.methodType().stringValue().equals(inv.type().stringValue())) {
+
+                                                //store stacked method parameters into locals
+                                                var storeStack = new LinkedList<StoreInstruction>();
```

Comment:
FWIW you can use an ArrayDeque + push + forEach, in the spirit of highlighting Deque over LinkedList for stack-based usage (since this is an example with visibility).